src/main/java/weiweibot/storage/Storage.java

```diff
@@ -0,0 +1,158 @@
+package weiweibot.storage;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+import weiweibot.exceptions.WeiExceptions;
+import weiweibot.tasks.Deadline;
+import weiweibot.tasks.Event;
+import weiweibot.tasks.Task;
+import weiweibot.tasks.TaskList;
+import weiweibot.tasks.Todo;
+
+/**
+ * File-backed storage for tasks using a simple line-based format.
+ *
+ * <p>The path is constructed from the provided segments (relative and
+ * OS-independent). Each task is stored on one line and reconstructed on load.</p>
+ */
+public class Storage {
+    private static final DateTimeFormatter FILE_DT = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    private final Path file;
+
+    public Storage(String... pathSegments) {
+        this.file = Paths.get("", pathSegments); // relative + OS-independent
+    }
+
+    /**
+     * Loads tasks from the backing file if it exists.
+     *
+     * <p>Blank lines are ignored. If a line cannot be parsed, a {@link WeiExceptions}
+     * is thrown indicating the corrupted line number.</p>
+     *
+     * @return a {@link TaskList} containing the parsed tasks (empty if file absent)
+     * @throws WeiExceptions on I/O errors or malformed records
+     */
+    public TaskList load() {
+        List<Task> list = new ArrayList<>();
```

Comment:
Maybe a more descriptive/less generic name than "list" might be appropriate here, to avoid confusion